packages/data/src/components/use-select/test/index.js

```diff
@@ -203,4 +203,79 @@ describe( 'useSelect', () => {
 			}
 		);
 	} );
+
+	it( 'uses memoized selector if dependent stores do not change', async () => {
+		const storeConfig = {
+			actions: {
+				increment: () => ( { type: 'INCREMENT' } ),
+			},
+			reducer: ( state, action ) => {
+				if ( ! state ) {
+					return { counter: 0 };
+				}
+				if ( action?.type === 'INCREMENT' ) {
+					return { counter: state.counter + 1 };
+				}
+				return state;
+			},
+			selectors: {
+				getCounter: ( state ) => state.counter,
+			},
+		};
+		registry.registerStore( 'store-1', storeConfig );
+		registry.registerStore( 'store-2', storeConfig );
+
+		const selectSpyFoo = jest
+			.fn()
+			.mockImplementation( ( { getCounter } ) => getCounter() );
+		const selectSpyBar = jest.fn().mockImplementation( () => 2 );
+		const TestComponent = jest.fn().mockImplementation( () => {
+			const data = useSelect( 'store-1', selectSpyFoo, [] );
+			useSelect( selectSpyBar, [] );
+			return <div>{ 'Counter: ' + data }</div>;
+		} );
+		let renderer;
+		act( () => {
+			renderer = TestRenderer.create(
```

Comment:
Nitpick: I'd prefer to just use RTL, but this works too.

Reply:
Yeah, I just found out that other files here use `react-test-renderer` as well. I agree to use `react-test-renderer` as is for simplicity.

Reply:
> RTL uses react-test-renderer as well.

For the most part, I think RTL mainly just uses `ReactDOM.render` but not `react-test-renderer`, but I got your point 😂